src/open_inwoner/cms/cases/views/status.py

```diff
@@ -290,74 +388,112 @@ def handle_document_upload(self, request, form):
+                self.log_system_action(
+                    "created klant for basic authenticated user",
+                    user=self.request.user,
+                )
+            else:
+                self.log_system_action(
+                    "could not create klant for BSN-user", user=self.request.user
+                )
 
-        # check if this info_object belongs to this case
-        if not fetch_case_information_objects_for_case_and_info(
-            self.case.url, info_object.url
-        ):
-            raise PermissionDenied()
+        # create contact moment
+        contact_moment = form.cleaned_data["contact_moment"]
+        data = {
+            "bronorganisatie": config.register_bronorganisatie_rsin,
+            "tekst": contact_moment,
+            "type": config.register_type,
+            "kanaal": "Internet",
+            "medewerkerIdentificatie": {
+                "identificatie": config.register_employee_id,
+            },
+        }
 
-        # check if this info_object should be visible
-        config = OpenZaakConfig.get_solo()
-        if not is_info_object_visible(info_object, config.document_max_confidentiality):
-            raise PermissionDenied()
+        contactmoment = create_contactmoment(data, klant=klant)
 
-        # retrieve and stream content
-        content_stream = download_document(info_object.inhoud)
-        if not content_stream:
-            raise Http404
+        if contactmoment:
+            messages.add_message(self.request, messages.SUCCESS, _("Vraag verstuurd!"))
+            self.log_system_action(f"registered contactmoment by API")
+        else:
+            messages.add_message(
+                self.request, messages.ERROR, _("Probleem bij versturen van de vraag.")
+            )
+            self.log_system_action(f"error while registering contactmoment by API")
 
-        self.log_user_action(
-            self.request.user,
-            _("Document van zaak gedownload {case}: {filename}").format(
-                case=self.case.identificatie,
-                filename=info_object.bestandsnaam,
-            ),
+        return HttpResponseClientRedirect(
+            reverse("cases:case_detail", kwargs={"object_id": str(self.case.uuid)})
         )
 
-        headers = {
-            "Content-Disposition": f'attachment; filename="{info_object.bestandsnaam}"',
-            "Content-Type": info_object.formaat,
-            "Content-Length": info_object.bestandsomvang,
-        }
-        response = StreamingHttpResponse(content_stream, headers=headers)
-        return response
-
-    def handle_no_permission(self):
-        # plain error and no redirect
-        raise PermissionDenied()
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["contact_form"] = self.get_form()
+        context["hxpost_contact_form"] = reverse(
+            "cases:case_detail_contact_form", kwargs=self.kwargs
+        )
```

Comment:
Not 100% sure the naming is great: 

If `contact_form` has a `_form` postfix because it holds a form, then `hxpost_contact_form` suggests it is also a form? 

Same for `hxpost_document_form` earlier. Maybe it should be `_action` because it sort-of does what the regular form action attributes does?